File: models/Diet.js
import mongoose from "mongoose";

const dietSchema = new mongoose.Schema({
  userId: mongoose.Schema.Types.ObjectId,
  bmi: Number,
  goal: String,
  dailyProteinTarget: Number,

  meals: {
    dailyProteinTarget: Number,
    breakfast: [mongoose.Schema.Types.Mixed],  // Can be string or object with meal and protein
    lunch: [mongoose.Schema.Types.Mixed],
    snacks: [mongoose.Schema.Types.Mixed],
    dinner: [mongoose.Schema.Types.Mixed],
  },

  createdAt: { type: Date, default: Date.now }
});

export default mongoose.model("Diet", dietSchema);

File: models/diet.py
from typing import Optional, List, Any
from datetime import datetime
from beanie import Document
from pydantic import BaseModel


class MealsData(BaseModel):
    dailyProteinTarget: Optional[float] = None
    breakfast: Optional[List[Any]] = []
    lunch: Optional[List[Any]] = []
    snacks: Optional[List[Any]] = []
    dinner: Optional[List[Any]] = []


class Diet(Document):
    userId: str  # ObjectId as string
    bmi: Optional[float] = None
    goal: Optional[str] = None
    dailyProteinTarget: Optional[float] = None
    meals: Optional[MealsData] = None
    createdAt: datetime = datetime.now()

    class Settings:
        name = "diets"
